packages/lodestar/src/db/repositories/blockArchive.ts

```diff
@@ -101,14 +101,21 @@ export class BlockArchiveRepository extends Repository<Slot, allForks.SignedBeac
     ]);
   }
 
+  async *p2pBlockStream(opts?: IBlockFilterOptions): AsyncIterable<P2pBlockResponse> {
+    const firstSlot = this.getFirstSlot(opts);
+    const binaryValuesStream = super.binaryValuesStream(opts);
```

Comment:
You can stream entries as bytes and get the slot from the key and well as the value in bytes